src/sortStudents.ts

```diff
@@ -1,16 +1,63 @@
 
 export interface Student {
-  // describe Student interface
+  'name': string,
+  'surname': string,
+  'age': number,
+  'married': boolean,
+  'grades': number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  type CallBack = (student1: Student, student2: Student) => number;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function calculateAverageGrade(student:Student): number {
+    return student.grades.reduce((sum: number, n: number) => sum + n, 0)
+    / student.grades.length;
+  }
+
+  const callback: CallBack = (student1, student2) => {
+    switch (sortBy) {
+      case SortType.Name:
+        return order === 'asc'
+          ? student1[sortBy].localeCompare(student2[sortBy])
+          : student2[sortBy].localeCompare(student1[sortBy]);
+
+      case SortType.Surname:
+        return order === 'asc'
+          ? student1[sortBy].localeCompare(student2[sortBy])
+          : student2[sortBy].localeCompare(student1[sortBy]);
```

Comment:
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc'
          ? student1[sortBy].localeCompare(student2[sortBy])
          : student2[sortBy].localeCompare(student1[sortBy]);
```
you can group cases in such way